Source/Cosmology/CosmoMovingPunctureGauge.hpp

```diff
@@ -0,0 +1,77 @@
+/* GRChombo
+ * Copyright 2012 The GRChombo collaboration.
+ * Please refer to LICENSE in GRChombo's root directory.
+ */
+
+#ifndef COSMOMOVINGPUNCTUREGAUGE_HPP_
+#define COSMOMOVINGPUNCTUREGAUGE_HPP_
+
+#include "DimensionDefinitions.hpp"
+#include "MovingPunctureGauge.hpp"
+#include "Tensor.hpp"
+
+/// This is an example of a gauge class that can be used in the CCZ4RHS compute
+/// class
+/**
+ * This class implements a slightly more generic version of the moving puncture
+ * gauge. In particular it uses a Bona-Masso slicing condition of the form
+ * f(lapse) = -c*lapse^(p-2)
+ * and a Gamma-driver shift condition
+ **/
+class CosmoMovingPunctureGauge
+{
+  public:
+    using params_t = MovingPunctureGauge::params_t;
+    /*
+    struct params_t
+    {
+        // lapse params:
+        double lapse_advec_coeff = 0.; //!< Switches advection terms in
+                                       //! the lapse condition on/off
+        double lapse_power = 1.; //!< The power p in \f$\partial_t \alpha = - c
+                                 //!\alpha^p(K-2\Theta)\f$
+        double lapse_coeff = 2.; //!< The coefficient c in \f$\partial_t \alpha
+                                 //!= -c \alpha^p(K-2\Theta)\f$
+        // shift params:
+        double shift_Gamma_coeff = 0.75; //!< Gives the F in \f$\partial_t
+                                         //!  \beta^i =  F B^i\f$
+        double shift_advec_coeff = 0.;   //!< Switches advection terms in the
+                                         //! shift condition on/off
+        double eta = 1.; //!< The eta in \f$\partial_t B^i = \partial_t \tilde
+                         //!\Gamma - \eta B^i\f$
+    };
+    */
+
+  protected:
+    params_t m_params;
+    double m_K_mean;
+
+  public:
+    CosmoMovingPunctureGauge(const params_t &a_params) : m_params(a_params) {}
```

Comment:
I think you should pass K_mean in the constructor, even if you default it to zero, as otherwise you risk evolving with it unset.